sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultTaskEnvRouter.java

```diff
@@ -0,0 +1,127 @@
+package com.mesosphere.sdk.config;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.mesosphere.sdk.offer.taskdata.EnvUtils;
+
+/**
+ * Default implementation of a {@link TaskEnvRouter} which handles forwarding environment variables from two sources:
+ * <ul>
+ * <li>Scheduler's system environment: searches for environment variables that start with {@code "TASKCFG_<TASKTYPE>_"}
+ * and/or {@code "TASKCFG_ALL_"}, and returns appropriate {@link ConfigNamespace}s for task types based on that
+ * environment.</li>
+ * <li>Manually-provided values: The service developer may provide custom values which will then be forwarded to tasks
+ * as needed.</li>
+ * </ul>
+ */
+public class DefaultTaskEnvRouter implements TaskEnvRouter {
```

Comment:
I don't see a compelling reason to have an Interface in front of this?

Is it just to limit what we see elsewhere to just the output of the router, and not expose the inputs?